proposals/NNNN-vehicle-type-filter.md

```diff
@@ -0,0 +1,435 @@
+7. If `Get vehicle type` message response is NACK, the proxy will end RPC session and the app will not be registered on the SDL system.
+
+![Sequence Diagram](../assets/proposals/NNNN-vehicle-type-filter/ios.png)
+
+## Potential downsides
+
+The Android SDL router service after transport connection needs to start RPC service. If the SDL router service does not start RPC service and relies on the client to send the `Get vehicle type` protocol message, the connected client would start another foreground service. This would force the connected app to show two notifications to users while determining vehicle type information and will need to terminate services if the app does not support SDL enabled IVI system. 
+
+## Impact on existing code
+
+Above mentioned changes need to be implemented in SDL Core, the android proxy and iOS proxy.
+
+## Alternatives considered
+
+### GetVehicleType as an RPC instead of Protocol message
+1. As per current implementation, the RAI is the first RPC sent to SDL enabled IVI system and the vehicle type information is available to the app in its response. 
+2. If `GetVehicleType` is an RPC, this behavior needs to be changed. The app should be able to use `GetVehicleType` RPC before `RegisterAppInterface` RPC.
+3. Since the RPC message version is sent to the app in `RAI response`, the app, when connected to older SDL enabled IVI, would not know if `GetVehicleType` RPC is supported or not.
+4. If `GetVehicleType` is an RPC and it sent first, the app would need to depend on timeout to know the support for `GetVehicleType` RPC when connected to older SDL enable IVI system. This behavior would add delays in app registration on the SDL system.
+5. If `GetVehicleType` is an RPC, the exclusive apps could potentially show two notifications described above for Android apps before terminating them. Terminating router service hosted by exclusive apps will unregister all apps using it.
+
+### Soft-Registrations of an App
+1. The app sends the RAI Request and receives its Response.
+2. The HMI does not register the app until the app completes its validation.
+3. On completing validation, the app sends some indication for hard app registration. The app will be shown on SDL to enable the IVI system. The type of notification to the SDL enabled IVI system needs to be defined.
+4. This solution would solve the proprietary app registrations. In this case, as well, the exclusive apps could potentially show two notifications described above for Android apps before terminating them. Terminating router service hosted by exclusive apps will unregister all apps using it.
+
+### Creating Generic SDL App to create SDL router service
+
+1. The user will need to download an additional app for the SDL connection.
+2. This would affect App discoverability, as the users would potentially never know about such app and never find apps in SDL enabled system. 
```

Comment:
```suggestion
2. This would affect App discoverability, as the users would potentially never know about such app and never find apps in an SDL enabled system. 
```